### PR TITLE
ITs for logtest verbose mode added

### LINKED ISSUE #1584: ITs logtest use of verbose mode
Hello team,

In order to test the new feature of Logtest that allows it to use the `verbose` mode to debug the rules (see [Wazuh-logtest: verbose mode #9204](https://github.com/wazuh/wazuh/pull/9204), we will create the following Integration tests:

- [x] Test rule match case if the `rules_debug` option is omitted
- [x] Test rule match case  if the `rules_debug` option is set to `true`
- [x] Test rule match case  if the `rules_debug` option is set to `false`
- [x] Test rule match case  if the `rules_debug` option is erroneous (string)
- [x] Test rule match case  if the `rules_debug` option is erroneous (number)
- [x] Test rule match case  if the `rules_debug` option is erroneous (object)
- [x] Test rule match case  if the `rules_debug` option is erroneous (array)
- [x] Test rule match case  if the `options` field is erroneous (boolean)
- [x] Test rule match case  if the `options` field is erroneous (array)
- [x] Test rule match case  if the `options` field is erroneous (number)
- [x] Test rule match case  if the `options` field is erroneous (string)

Regards,
Juan Cabrera

### PR DESCRIPTION
**Pull request by nmkoremblum**
|Related issue|
|---|
| close #1584 |

## Description

Hi team,

This PR aims to close the issue #1584. As for the [new Logtest feature](https://github.com/wazuh/wazuh/issues/7779), that allows it to use the verbose mode to debug the rules (see Wazuh-logtest: verbose mode #9204, we have created the following Integration Tests:

- [x] Test rule match case if the `rules_debug` option is omitted
- [x] Test rule match case  if the `rules_debug` option is set to `true`
- [x] Test rule match case  if the `rules_debug` option is set to `false`
- [x] Test rule match case  if the `rules_debug` option is erroneous (string)
- [x] Test rule match case  if the `rules_debug` option is erroneous (number)
- [x] Test rule match case  if the `rules_debug` option is erroneous (object)
- [x] Test rule match case  if the `rules_debug` option is erroneous (array)
- [x] Test rule match case  if the `options` field is erroneous (boolean)
- [x] Test rule match case  if the `options` field is erroneous (array)
- [x] Test rule match case  if the `options` field is erroneous (number)
- [x] Test rule match case  if the `options` field is erroneous (string)

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.
- [x] The test is documented in wazuh-qa/docs.
- [ ] `provision_documentation.sh` generate the docs without errors.

### CODE REVIEW

**Comment by JcabreraC** [comment]
## Test
```
# python3 -m pytest tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py -vvs
======================================== test session starts ========================================
platform linux -- Python 3.6.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.6.9', 'Platform': 'Linux-5.4.0-77-generic-x86_64-with-Ubuntu-18.04-bionic', 'Packages': {'pytest': '6.2.3', 'py': '1.10.0', 'pluggy': '0.13.1'}, 'Plugins': {'html': '3.1.1', 'testinfra': '5.0.0', 'metadata': '1.9.0'}}
rootdir: /home/Wazuh/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-1.9.0
collected 11 items                                                                                  

tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py::test_rules_verbose[The rules_debug field is omitted] PASSED
tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py::test_rules_verbose[The rules_debug field is set to true] PASSED
tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py::test_rules_verbose[The rules_debug field is set to false] PASSED
tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py::test_rules_verbose[The rules_debug field is set to a bad type (string)] PASSED
tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py::test_rules_verbose[The rules_debug field is set to a bad type (number)] PASSED
tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py::test_rules_verbose[The rules_debug field is set to a bad type (object)] PASSED
tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py::test_rules_verbose[The rules_debug field is set to a bad type (array)] PASSED
tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py::test_rules_verbose[The options field is set to a bad type (boolean)] PASSED
tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py::test_rules_verbose[The options field is set to a bad type (array)] PASSED
tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py::test_rules_verbose[The options field is set to a bad type (number)] PASSED
tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py::test_rules_verbose[The options field is set to a bad type (string)] PASSED

======================================== 11 passed in 10.60s ========================================
```

File: tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+
+@pytest.fixture(scope='module', params=test_cases, ids=[test_case['name'] for test_case in test_cases])
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+# Tests
+def test_rules_verbose(get_configuration, configure_rules_list, connect_to_sockets_function):
```
Comment (Rebits) [inline tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py:62]: This test did not use any fixture to up necessary wazuh modules. This makes this test depends on the rest of the suite.

Please create or use an existing fixture to up minimum modules to use for this test module.

**Review by Rebits — CHANGES_REQUESTED** [review]

File: tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py (hunk 1 above)
Comment (Rebits) [reply tests/integration/test_logtest/test_log_process_options/test_rules_verbose.py:62]: In this case, this test only needs to up required daemons. So we can create the following fixture:
```
@pytest.fixture(scope='module')
def restart_required_logtest_daemons(request):
    """Wazuh daemons handler.
    """
    required_logtest_daemons = ['wazuh-analysisd', 'wazuh-monitord']


    for daemon in required_logtest_daemons:
        control_service('restart', daemon=daemon)

    for i in required_logtest_daemons:
        check_daemon_status(daemon)

    yield 

    for i in required_logtest_daemons:
        control_service('stop', daemon=daemon)

```

And calls it in the test

```
# Tests
def test_rules_verbose(restart_required_logtest_daemons, get_configuration, configure_rules_list, 
                       connect_to_sockets_function):
```

**Review by Rebits — CHANGES_REQUESTED** [review]
Minor changes in daemons handling are required

**Review by Rebits — CHANGES_REQUESTED** [review]
GJ but minor changes are required.

**Comment by Rebits** [comment]
| Local | Status 	| Report 	|
|---	|---	|---	|
| R1 	| :green_circle:  	|  	[R1](https://github.com/wazuh/wazuh-qa/files/7196645/R1.zip)|
| R2 	| :green_circle: 	|  	[R2](https://github.com/wazuh/wazuh-qa/files/7196646/R2.zip)|
| R3 	| :green_circle: 	|  	 [R3](https://github.com/wazuh/wazuh-qa/files/7196647/R3.zip)   |

**Review by Rebits — DISMISSED** [review]
LGTM

**Comment by Rebits** [comment]
| Local | Status 	| Report 	|
|---	|---	|---	|
| R1 	| :green_circle:  	|  	[R1](https://github.com/wazuh/wazuh-qa/files/7217281/R1.zip)|
| R2 	| :green_circle: 	|  	[R2](https://github.com/wazuh/wazuh-qa/files/7217282/R2.zip)|
| R3 	| :green_circle: 	|  	 [R3](https://github.com/wazuh/wazuh-qa/files/7217283/R3.zip)|

**Review by Rebits — APPROVED** [review]

**Pull request merged**